sdk/storage/storage-file-datalake/src/DataLakeDirectoryClient.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { CanonicalCode } from "@opentelemetry/types";
+
+import { DataLakeFileClient } from "./DataLakeFileClient";
+import { DataLakePathClient } from "./DataLakePathClient";
+import {
+  DirectoryCreateOptions,
+  DirectoryCreateResponse,
+  PathCreateOptions,
+  PathCreateResponse,
+  PathResourceType,
+} from "./models";
+import { createSpan } from "./utils/tracing";
+import { appendToURLPath } from "./utils/utils.common";
+
+export class DataLakeDirectoryClient extends DataLakePathClient {
+  // https://stackoverflow.com/questions/50729485/override-method-with-different-argument-types-in-extended-class-typescript
+  public async create(
+    resourceType: PathResourceType,
+    options?: PathCreateOptions
+  ): Promise<PathCreateResponse>;
+  public async create(options?: DirectoryCreateOptions): Promise<DirectoryCreateResponse>;
+  public async create(
+    resourceTypeOrOptions?: PathResourceType | PathCreateOptions,
```

Comment:
DirectoryCreateOptions?

Reply:
DirectoryCreateOptions is an alias of PathCreateOptions

Reply:
Will update